config/confighttp/confighttp.go

```diff
@@ -107,10 +107,12 @@ type ClientConfig struct {
 // We encourage to use this function to create an object of ClientConfig.
 func NewDefaultClientConfig() ClientConfig {
 	// The default values are taken from the values of 'DefaultTransport' of 'http' package.
+
 	maxIdleConns := 100
 	idleConnTimeout := 90 * time.Second
 
 	return ClientConfig{
+		Headers:         make(map[string]configopaque.String),
```

Comment:
We should also use configtls.NewDefaultClientConfig and configauth.NewDefaultAuthentication